src/Security/Authentication/test/CertificateTests.cs

```diff
@@ -929,39 +932,110 @@ public async Task VerifyValidationResultNeverCachedAfter30Min(bool cache)
 
     private static class Certificates
     {
-        public static X509Certificate2 SelfSignedPrimaryRoot { get; private set; } =
-            new X509Certificate2(GetFullyQualifiedFilePath("validSelfSignedPrimaryRootCertificate.cer"));
+        private static string ServerEku = "1.3.6.1.5.5.7.3.1";
+        private static string ClientEku = "1.3.6.1.5.5.7.3.2";
 
-        public static X509Certificate2 SignedSecondaryRoot { get; private set; } =
-            new X509Certificate2(GetFullyQualifiedFilePath("validSignedSecondaryRootCertificate.cer"));
+        static Certificates()
+        {
+            DateTimeOffset now = DateTimeOffset.UtcNow;
+
+            SelfSignedPrimaryRoot = MakeCert(
+                "CN=Valid Self Signed Client EKU,OU=dev,DC=idunno-dev,DC=org",
+                ClientEku,
+                now);
+
+            SignedSecondaryRoot = MakeCert(
+                "CN=Valid Signed Secondary Root EKU,OU=dev,DC=idunno-dev,DC=org",
+                ClientEku,
+                now);
+
+            SelfSignedValidWithServerEku = MakeCert(
+                "CN=Valid Self Signed Server EKU,OU=dev,DC=idunno-dev,DC=org",
+                ServerEku,
+                now);
+
+            SelfSignedValidWithClientEku = MakeCert(
+                "CN=Valid Self Signed Server EKU,OU=dev,DC=idunno-dev,DC=org",
+                ClientEku,
+                now);
+
+            SelfSignedValidWithNoEku = MakeCert(
+                "CN=Valid Self Signed No EKU,OU=dev,DC=idunno-dev,DC=org",
+                eku: null,
+                now);
+
+            SelfSignedExpired = MakeCert(
+                "CN=Expired Self Signed,OU=dev,DC=idunno-dev,DC=org",
+                eku: null,
+                now.AddYears(-2),
+                now.AddYears(-1));
+
+            SelfSignedNotYetValid = MakeCert(
+                "CN=Not Valid Yet Self Signed,OU=dev,DC=idunno-dev,DC=org",
+                eku: null,
+                now.AddYears(2),
+                now.AddYears(3));
+
+            SignedClient = MakeCert(
+                "CN=Valid Signed Client,OU=dev,DC=idunno-dev,DC=org",
+                ClientEku,
+                now);
```

Comment:
From the name, and usage, I'd expect that SignedClient is supposed to be issued by SelfSignedPrimaryRoot  (or maybe SelfSignedSecondaryRoot?).  But it's just another self-signed cert.

Reply:
You are right:

![image](https://user-images.githubusercontent.com/6537861/151427451-d76b9e94-93ee-4802-a51e-c76215e75e85.png)

So would it be correct for me to just slap on a "issuer CN=Valid Signed Secondary Root" to match the old cert?

Reply:
No.  You'll have to, for this cert, replace `return request.CreateSelfSigned(notBefore, notAfter);`

```C#
Span<byte> serial = stackalloc byte[19];
RandomNumberGenerator.Fill(serial);

using (X509Certificate2 pubOnly = request.Create(SelfSignedSecondaryRoot, notBefore, notAfter, serial))
{
    return pubOnly.CopyWithPrivateKey(key);
}
```


So, something like

```C#
        private static X509Certificate2 MakeCert(
            string subjectName,
            string eku,
            DateTimeOffset notBefore,
            DateTimeOffset notAfter,
            X509Certificate2 issuerCert = null)
        {
            using (var key = RSA.Create(2048))
            {
                CertificateRequest request = new CertificateRequest(
                    subjectName,
                    key,
                    HashAlgorithmName.SHA256,
                    RSASignaturePadding.Pkcs1);

                if (eku != null)
                {
                    request.CertificateExtensions.Add(
                        new X509EnhancedKeyUsageExtension(
                            new OidCollection { new Oid(eku, null) }, false));
                }

                if (issuerCert == null)
                {
                    return request.CreateSelfSigned(notBefore, notAfter);
                }
                else
                {
                    Span<byte> serial = stackalloc byte[19];
                    RandomNumberGenerator.Fill(serial);

                    using (X509Certificate2 pubOnly = request.Create(SelfSignedSecondaryRoot, notBefore, notAfter, serial))
                    {
                        return pubOnly.CopyWithPrivateKey(key);
                    }
               }
        }
```

But you'll have to also create SelfSignedSecondaryRoot as a valid CA cert.  (An X509BasicConstraintsExtension where isCa is true, and the key usages needs to include cert signing)

Reply:
Ok I will give that a shot, but this leads me to ask, are we missing some tests to exercise these kinds of subtle cert differences, or it doesn't really matter to us since at our layer we are just passing the certs.  Basically if no tests fail without the issuer in the old cert, maybe we should either add tests to verify this, or don't bother specifying it... cc @blowdart for his thoughts as well

Reply:
I mean obviously switch away from my old cert names :)

Reply:
Well... none of your certs here have valid CN/SAN entries to be used as a TLS server, so there is clearly some custom validation going on wherever these certs are used.  I don't see where it's happening, so it's hard to say what's what.

https://github.com/dotnet/aspnetcore/blob/a5ac61c19228abb7755b54b57271addb4aa9f177/src/Security/Authentication/test/CertificateTests.cs#L347-L361

certainly looks like it should have failed in your current configuration (and the current configuration with checked-in certs... unless SelfSignedSecondaryRoot is actually a child issuing authority of SelfSignedPrimaryRoot?).  It feels like it should only pass when the server cert chains up to SelfSignedPrimaryRoot.

Of course, that test is disabled, but should presumably have had the Skip removed in this PR since it's addressing the issue of expired certs.

Reply:
Right, for some reason only VerifyExpiredSelfSignedFails and VerifyNotYetValidSelfSignedFails fail but only on Ubuntu right now